packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -1016,7 +1175,7 @@ class _ShortcutsState extends State<Shortcuts> {
   void initState() {
     super.initState();
     if (widget.manager == null) {
-      _internalManager = ShortcutManager();
+      _internalManager = ShortcutManager(useRawKeyEvents: false);
```

Comment:
How/when would we switch this over to true?

Since most people don't actually provide their own ShortcutManager, the new code path would never run for them until we flip this flag. I wonder what the migration path for people would be here?

If we merge this PR as-is, people would see deprecation warnings on their overrides of `ShortcutActivator.accepts` and `ShortcutManager.handleKeypress`. They will hopefully respond by adding the corresponding new methods ending with `*Event` to their implementations, which (as long as we keep this flag set to false) will actually never run. Even worse: They cannot actually remove the deprecated overrides from their code base (and probably must ignore the now unactionable analyzer deprecation warnings to make CI happy) because that is the code that still runs while we have `useKeyEvents` set to false. At some point we would switch `useRawKeyEvents` to true and run their new code. Unfortunately, we have no way of now telling people that they can now safely remove the old code short of us removing the methods from the superclasses altogether and breaking them for overriding a non-existent method...

I don't like that this will encourage people to add code that (for now) never runs while also forcing them to keep the deprecated old code around until further notice. Ideally, I can silence a deprecation warning in my code base by migrating off the deprecated stuff...

We are in a tough spot here. I wonder what the impact of a hard break would be (i.e. we just change the parameter type of the existing methods to the new Event classes)? Looking at google3, it would break 2 instances for `handleKeypress` and 3 for `accepts` and nothing for `isActivatedBy`. That's not nothing... but they could be "easily" migrated (and by changing the type to `dynamic` in the override and using a type check package authors could even stay backwards compatible for the duration of the deprecation period, if they care about that sort of thing).

I really don't like hard breaks that impact any customer, but maybe for this particular case it is the better choice given the alternatives? Another thing to still check would be to see how our other non-google3 customer tests would respond to such a hard break (if the impact is small we could work with those packages authors (i.e. send them PRs) to migrate to something that remains backwards compatible as outlined above, though).

Reply:
You have good points there.  Maybe we should just break things.  That means that we can keep the names of the functions, at least.

My thoughts in doing it the way it is: I was hoping that making all the internally allocated instances of `ShortcutManager` turn on `useRawKeyEvents` would provide enough cases where it was being used (it runs in all regularly constructed `Shortcuts` widgets, for instance). `ShortcutManager`, and the base `ShortcutActivator` classes are rarely subclassed. It seemed to me like the only instances that were in the state you describe were people who subclass `ShortcutManager` and those who wrote direct subclasses of `ShortcutActivator` instead of subclassing one of the more popular subclasses (e.g. `SingleActivator` and `CharacterActivator`). I think these are small groups, but perhaps I'm wrong about that.

However, if you think we would be better off breaking things, then maybe that's just what we should do.  I know that if I ask myself "what is the best end-state API", I'd rather just keep `accepts` and `isActivatedBy` (although Tong has some good points about maybe being able to eliminate `isActivatedBy`), and avoid having a period where both `accepts` and `acceptsEvent` exist to confuse people.

Reply:
In my comment I mistakingly assumed that the default for internally constructed ShortcutManagers are the old events, when in fact it's defaulting to the new events (unless you construct your own ShortcutManager). 

Wouldn't this mean, though, that people overriding ShortcutActivator.accepts and using it with a regular Shortcuts widget (which internally constructs a ShortcutManager defaulting to the new methods) would be instantly broken? Sure, they see a deprecation method on their override, but it would in fact no longer be called? Unless we can make the default implementation of ShortcutActiviator.acceptsEvent somehow call through to ShortcutActivator.accepts?

People overriding ShortcutsManager.handleKeyPress are a bit better off. Since they have to manually construct the ShortcutManager, they are in control of whether old or new methods are called. Once they implement the new methods, they'd have to use the deprecated property useRawKeyEvents to switch over, though, and then be "broken" when we remove it.

Reply:
Yes, good points, I think that we're probably better off just doing the hard break, then.

I do think that there aren't a lot of people who override these things, but those people would be broken in a very subtle way.  And that also argues that a "hard break" won't affect too many people directly anyhow.